### DISCUSSION TITLE
Strange behaviour with authRefresh in production

### DISCUSSION THREAD

**Discussion by djt** [pocketbase/pocketbase#3126] [Q&A]
I'm sure this is unlikely to be a pocketbase bug, but I am having an issue moving to prod with my current SSR implementation.  A few details:

NextJS is running via next start in prod.
It is reverse proxied via Caddy.
When running locally, it all worked fine, but there was no proxying.

The issue I'm having is in my middleware, the cookie is loaded, and seen as valid by the authStore, but authRefresh is throwing and exception with the following error (can confirm that it is also being seen as 'guest' in the request log within pocketbase):

```
{
  "errorDetails": "<nil>",
  "errorMessage": "The request requires valid record authorization token to be set."
}
```

The basic code in question is:

    
```
  try {
      pb.authStore.loadFromCookie(cookie)
      console.log('Auth Model', pb.authStore.model) // Shows record
      console.log('Is Valid', pb.authStore.isValid) // True
      pb.authStore.isValid && await pb.collection('users').authRefresh()
  } catch (e) {
      pb.authStore.clear()
      response.cookies.delete('pb_auth')
      return NextResponse.redirect(new URL('/login?error=NO_AUTH', request.url))
  }
```

If you could provide any insight into how I can get this working it would be much appreciated.

**Comment by ganigeorgiev** [comment]
`pb.authStore.isValid` only loosely checks whether the token is valid or not (eg. has a parsable payload and non-expired `exp` claim), aka. it doesn't send request to the server to verify it.

Unfortunately I'm not sure what could be the cause of the issue.
There were several reported issues with Next.js but all so far were related to the default enabled caching (eg. https://github.com/pocketbase/pocketbase/discussions/2695, https://github.com/pocketbase/pocketbase/discussions/3072).
In this case I don't think it is cache because `authRefresh()` sends a `POST` request and caching should be disabled by default for such requests, so I'm out of ideas.

If you think that it is caused by something in the JS SDK or PocketBase, please create a **minimal** reproducible repo and I'll try to investigate it.
